src/main/resources/assets/js/components/AppPageComponent.jsx

```diff
@@ -8,66 +9,78 @@ var TaskDetailComponent = require("../components/TaskDetailComponent");
 var TaskViewComponent = require("../components/TaskViewComponent");
 var TogglableTabsComponent = require("../components/TogglableTabsComponent");
 
-var tabs = [
-  {id: "tasks", text: "Tasks"},
-  {id: "configuration", text: "Configuration"}
+var tabsTemplate = [
+  {id: "apps/:appid", text: "Tasks"},
+  {id: "apps/:appid/configuration", text: "Configuration"}
 ];
 
 var AppPageComponent = React.createClass({
   displayName: "AppPageComponent",
 
   propTypes: {
-    activeTask: React.PropTypes.object,
     appVersionsFetchState: React.PropTypes.number.isRequired,
     model: React.PropTypes.object.isRequired,
     destroyApp: React.PropTypes.func.isRequired,
     fetchTasks: React.PropTypes.func.isRequired,
     fetchAppVersions: React.PropTypes.func.isRequired,
-    onDestroy: React.PropTypes.func.isRequired,
-    onShowTaskDetails: React.PropTypes.func.isRequired,
-    onShowTaskList: React.PropTypes.func.isRequired,
     onTasksKilled: React.PropTypes.func.isRequired,
     restartApp: React.PropTypes.func.isRequired,
     rollBackApp: React.PropTypes.func.isRequired,
     scaleApp: React.PropTypes.func.isRequired,
     suspendApp: React.PropTypes.func.isRequired,
-    tasksFetchState: React.PropTypes.number.isRequired
+    tasksFetchState: React.PropTypes.number.isRequired,
+    view: React.PropTypes.string
   },
 
   getInitialState: function () {
+    var appId = this.props.model.get("id");
+    var activeTabId;
+
+    var tabs = _.reduce(tabsTemplate, function (current, tab) {
```

Comment:
This should be a map, since you are outputting the same list, just changed:
```
var tabs = _.map(tabsTemplate, function (tab) {
      var id = tab.id.replace(":appid", encodeURIComponent(appId));
      if (activeTabId == null) {
        activeTabId = id;
      }

      return {
        id: id,
        text: tab.text
      };
    });
```